pkg/alertmanager/alertmanager.go

```diff
@@ -143,3 +158,83 @@ func (s Service) configure(ctx context.Context, alertmanagerConfigContent []byte
 
 	return nil
 }
+
+// deleteTenantConfiguration deletes a tenant using the to Mimir Alertmanager's API
+func (s Service) deleteTenantConfiguration(ctx context.Context, tenantID string) error {
+	logger := log.FromContext(ctx)
+
+	url := s.alertmanagerURL + alertmanagerAPIPath
+	logger.WithValues("url", url, "tenant", tenantID).Info("Alertmanager: deleting tenant")
+
+	// Send request to Alertmanager's API
+	req, err := http.NewRequest(http.MethodDelete, url, nil)
+	if err != nil {
+		return errors.WithStack(fmt.Errorf("alertmanager: failed to create request: %w", err))
+	}
+
+	req.Header.Set(common.OrgIDHeader, tenantID)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return errors.WithStack(fmt.Errorf("alertmanager: failed to send request: %w", err))
+	}
+	defer resp.Body.Close() // nolint: errcheck
+
+	logger.WithValues("status_code", resp.StatusCode).Info("Alertmanager: tenant configurationdeleted")
+
+	if resp.StatusCode != http.StatusCreated {
+		respBody, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return errors.WithStack(fmt.Errorf("alertmanager: failed to read response: %w", err))
+		}
+
+		e := APIError{
+			Code:    resp.StatusCode,
+			Message: string(respBody),
+		}
+
+		return errors.WithStack(fmt.Errorf("alertmanager: failed to delete tenant configuration: %w", e))
+	}
+
+	return nil
+}
+
+// deleteRules deletes all rules for a tenant using the to Mimir Alertmanager's API
+func (s Service) deleteRules(ctx context.Context, tenantID string) error {
+	logger := log.FromContext(ctx)
+
+	url := "http://mimir-gateway.mimir.svc/ruler/delete_tenant_config"
+	logger.WithValues("url", url, "tenant", tenantID).Info("Alertmanager: delete rules for tenant")
+
+	// Send request to Alertmanager's API
+	req, err := http.NewRequest(http.MethodPost, url, nil)
+	if err != nil {
+		return errors.WithStack(fmt.Errorf("alertmanager: failed to create request: %w", err))
+	}
+
+	req.Header.Set(common.OrgIDHeader, tenantID)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return errors.WithStack(fmt.Errorf("alertmanager: failed to send request: %w", err))
+	}
+	defer resp.Body.Close() // nolint: errcheck
+
+	logger.WithValues("status_code", resp.StatusCode).Info("Alertmanager: delete rules for tenant")
+
+	if resp.StatusCode != http.StatusCreated {
```

Comment:
I think the status returned here is 200 on success

![image](https://github.com/user-attachments/assets/e1240860-5340-42e9-9a39-ed2196fca07d)
https://grafana.com/docs/mimir/latest/references/http-api/#delete-tenant-configuration

Reply:
I forgot that created is 204 🤦🏻

Reply:
fixed